noir-projects/aztec-nr/aztec/src/macros/functions/interfaces.nr

```diff
@@ -176,7 +136,7 @@ pub comptime fn stub_fn(f: FunctionDefinition) -> Quoted {
                 name: $fn_name_str,
                 $args_hash
                 args: $args_acc_name,
-                original: $original,
+                return_type: std::mem::zeroed(),
```

Comment:
Call interfaces were sort of broken before. Return value types were unbound and that led to weird situations such as:

```noir
#[private]
my_fn() -> u8 {...}

...

#[test]
fn test_my_fn() {
  let result = MyContract::at(somewhere).my_fn().call(&mut env.private());
  assert(result == 8 // this is a Field, so result is cast to Field);
}
```

`return_value` in the call interfaces is essentially rust's `PhantomData`, so we are able to properly type the return value of the call interface (both in TXE and actual contracts)

Reply:
I thought I understood this, and did some reading around on `PhantomData`, but then tried to write down a comment explaining why `zeroed` needs to be there and failed.

Reply:
It's similar to when we needed array hints in methods for length...we need a return_type variable with the correct `Type`, but we don't know what the actual value is (because we haven't called the function yet)